### PR TITLE
Feature/key rotation

### PR DESCRIPTION
**Pull request by marcstreeter**
always streaming output for key rotation

### CODE REVIEW

**Comment by marcstreeter** [comment]
this could be simplified greater (and not commit the heresy of copy pasta) if the foundries/tuftree/client had a `go.mod` set up

**Comment by marcstreeter** [comment]
this PR is proposed instead of https://github.com/foundriesio/fioctl/pull/19

**Comment by doanac** [comment]
> this could be simplified greater (and not commit the heresy of copy pasta) if the foundries/tuftree/client had a go.mod set up

That's dead code. we don't want to use it like a module.

**Comment by marcstreeter** [comment]
> > this could be simplified greater (and not commit the heresy of copy pasta) if the foundries/tuftree/client had a go.mod set up
> 
> That's dead code. we don't want to use it like a module.

💀oh? I didn't realize.

File: cmd/keys.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+)
+
+var keysCmd = &cobra.Command{
+	Use:              "keys",
+	Short:            "Manage keys in use by your factory fleet",
+	PersistentPreRun: assertLogin,
```
Comment (doanac) [inline cmd/keys.go]: don't need that.

File: cmd/keys_rotate.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	return scriptPath, nil
+}
+
+func runRotationScript(imageName string, sourcePath string, credentialsPath string) error {
+	targetPath := "/tmp/tmp.py"
+	args := []string{
+		// base args
+		"run", "--rm",
+		// env args
+		"--env", "PYTHONUNBUFFERED=1",
```
Comment (doanac) [inline cmd/keys_rotate.go]: ah - i bet that was driving you mad.

Comment (marcstreeter) [reply cmd/keys_rotate.go]: it was 😄

Comment (marcstreeter) [reply cmd/keys_rotate.go]: quick bit of history, I initially tried using the tuft/client (dead) code  we've been talking about and it didn't change output at all (or so I thought at the time).  After switching back and trying with a Pipeline -- same issue, so I then just ran the command directly and noticed that docker wasn't outputting til the end.  After switching that environment variable - it was fixed.  Not with "CombinedOutput" though ( which I tried when doing this streaming version) -- hence the use again of this code. 

Took quite a tour to get here.  It was fun.

File: cmd/keys.go (hunk 1 above)
Comment (marcstreeter) [reply cmd/keys.go]: removed

**Comment by doanac** [comment]
I just toyed around with this and it works pretty well. Lets merge this and then we can circle back and get it using a directory rather than .tgz in the near future so that its more bullet-proof.

**Comment by marcstreeter** [comment]
> I just toyed around with this and it works pretty well. Lets merge this and then we can circle back and get it using a directory rather than .tgz in the near future so that its more bullet-proof.

k - just squashed all the commits before the merge to clean up. I'll merge now.

**Pull request merged**